### PR TITLE
Improve several areas of CurlHandler error handling

### PR DESCRIPTION
**Pull request by stephentoub**
While tracking down an issue related to exception handling in CurlHandler, I fixed up a few issues:
- On older libcurl's curl_multi_perform can return an error code that means to retry curl_multi_perform, but we were treating it as an actual error.
- Several places we were creating an HttpRequestException with a null inner exception, even if we had additional information about the cause.  I've fixed those call sites to include an appropriate inner exception.
- Cleaned up a few error messages.
- Fixed one place where I was overzealous in mapping exceptions to IOException

cc: @ericeil, @kapilash 
Fixes https://github.com/dotnet/corefx/issues/6268

### CODE REVIEW

File: src/System.Net.Http/src/Resources/Strings.resx [hunk 1]
```diff
@@ -298,16 +298,16 @@
     <value>ClientCertificateOption.Automatic is supported only with libcurl with libssl compatible backend.</value>
   </data>
   <data name="net_http_unix_invalid_credential" xml:space="preserve">
-    <value>libcurl does not support different credentials for different authentication schemes</value>
+    <value>the libcurl library in use does not support different credentials for different authentication schemes.</value>
```
Comment (davidsh) [inline src/System.Net.Http/src/Resources/Strings.resx]: Should "the libcurl library in use..." be "The libcurl library in use..." ?

Comment (stephentoub) [inline src/System.Net.Http/src/Resources/Strings.resx]: Oops, fixed.

**Comment by stephentoub** [comment]
Test Innerloop Ubuntu Debug Build and Test please

(Jenkins failure)

**Comment by stephentoub** [comment]
Test Innerloop Ubuntu Release Build and Test please

**Comment by stephentoub** [comment]
Test Innerloop OSX Debug Build and Test please

**Pull request merged**